File: clients/javascript/src/models/rateLimitCheckIn.ts
// this file is @generated

export interface RateLimitCheckIn {
    key: string;
    /** Number of tokens to consume (default: 1) */
    tokens?: number;
    /** Maximum capacity of the bucket */
    capacity: number;
    /** Number of tokens to add per refill interval */
    refillAmount: number;
    /** Interval in seconds between refills (minimum 1 second) */
    refillInterval?: number;
}

export const RateLimitCheckInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): RateLimitCheckIn {
        return {
            key: object['key'],
            tokens: object['tokens'],
            capacity: object['capacity'],
            refillAmount: object['refill_amount'],
            refillInterval: object['refill_interval'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: RateLimitCheckIn): any {
        return {
            'key': self.key,
            'tokens': self.tokens,
            'capacity': self.capacity,
            'refill_amount': self.refillAmount,
            'refill_interval': self.refillInterval,
        };
    }
}

File: clients/rust/src/models/rate_limit_check_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimitCheckIn {
    pub key: String,

    /// Number of tokens to consume (default: 1)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tokens: Option<u64>,

    /// Maximum capacity of the bucket
    pub capacity: u64,

    /// Number of tokens to add per refill interval
    pub refill_amount: u64,

    /// Interval in milliseconds between refills (minimum 1 millisecond)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub refill_interval_millis: Option<u64>,
}

impl RateLimitCheckIn {
    pub fn new(key: String, capacity: u64, refill_amount: u64) -> Self {
        Self {
            key,
            tokens: None,
            capacity,
            refill_amount,
            refill_interval_millis: None,
        }
    }

    pub fn with_tokens(mut self, value: impl Into<Option<u64>>) -> Self {
        self.tokens = value.into();
        self
    }

    pub fn with_refill_interval_millis(mut self, value: impl Into<Option<u64>>) -> Self {
        self.refill_interval_millis = value.into();
        self
    }
}
